### PR TITLE
Improve quick info on `using` keyword

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/75429

It turned out that quick info tests didn't contain any tests of `using` keyword at all, so I added a bunch

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs [hunk 1]
```diff
@@ -105,19 +105,64 @@ public SyntaxToken GenerateUniqueName(string baseName, IEnumerable<string> usedN
...
+
+            var currentType = type;
+            while (currentType is not null)
+            {
+                if (currentType
+                    .GetMembers(nameof(IAsyncDisposable.DisposeAsync))
+                    .FirstOrDefault(m => m is IMethodSymbol { DeclaredAccessibility: Accessibility.Public, ReturnType: var returnType, Parameters.Length: 0 } &&
+                                         SymbolEqualityComparer.Default.Equals(returnType, valueTaskType)) is { } disposeMethodFromPattern)
+                {
+                    return (IMethodSymbol)disposeMethodFromPattern;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs:136]:
```suggestion
                if (currentType
                    .GetMembers(nameof(IAsyncDisposable.DisposeAsync))
                    .FirstOrDefault(m => m is IMethodSymbol { DeclaredAccessibility: Accessibility.Public, ReturnType: var returnType, Parameters.Length: 0 } &&
                                         SymbolEqualityComparer.Default.Equals(returnType, valueTaskType)) is IMethodSymbol disposeMethodFromPattern)
                {
                    return disposeMethodFromPattern;
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs [hunk 2]
```diff
@@ -105,19 +105,64 @@ public SyntaxToken GenerateUniqueName(string baseName, IEnumerable<string> usedN
...
+            }
+        }
+        else
+        {
+            var currentType = type;
+            while (currentType is not null)
+            {
+                if (currentType
+                    .GetMembers(nameof(IDisposable.Dispose))
+                    .FirstOrDefault(m => m is IMethodSymbol { DeclaredAccessibility: Accessibility.Public, ReturnsVoid: true, Parameters.Length: 0 }) is { } disposeMethodFromPattern)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs:149]:
```suggestion
                    .FirstOrDefault(m => m is IMethodSymbol { DeclaredAccessibility: Accessibility.Public, ReturnsVoid: true, Parameters.Length: 0 }) is IMethodSymbol disposeMethodFromPattern)
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs [hunk 3]
```diff
@@ -105,19 +105,64 @@ public SyntaxToken GenerateUniqueName(string baseName, IEnumerable<string> usedN
...
+                {
+                    return (IMethodSymbol)disposeMethodFromPattern;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs:151]:
```suggestion
                    return disposeMethodFromPattern;
```

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SemanticsFactsService/AbstractSemanticFactsService.cs:151]: or, just use `OfType<IMethodSymbol>`

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Comment by DoctorKrolic** [comment]
Restart failed CI pipelines please

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi CI failed twice in a row

**Pull request merged**